code/game/objects/items/weapons/grenades/smokebomb.dm

```diff
@@ -37,3 +35,7 @@
 	sleep(80)
 	qdel(src)
 	return
+
+/obj/item/weapon/grenade/smokebomb/attackby(obj/item/I as obj, mob/user as mob)
+	if(istype(I,/obj/item/device/multitool))
+		smoke_color = input(user, "Choose a color for the smoke:") as color|null
```

Comment:
You want to check that the return value from input isn't null before assigning it to smoke_color then.
As a QoL thing you can supply smoke_color as the 4th argument to input and it'll be pre-marked during selection (the 3rd argument is the title).